lib/eth/client.rb

```diff
@@ -369,15 +369,24 @@ def reset_id
       @id = 0
     end
 
-    # Checkes wether a transaction is mined or not.
+    # Checks whether a transaction is mined or not.
     #
     # @param hash [String] the transaction hash.
     # @return [Boolean] true if included in a block.
-    def is_mined_tx?(hash)
+    def tx_mined?(hash)
       mined_tx = eth_get_transaction_by_hash hash
       !mined_tx.nil? && !mined_tx["result"].nil? && !mined_tx["result"]["blockNumber"].nil?
     end
 
+    # Checks whether a transaction succeeded or not.
+    #
+    # @param hash [String] the transaction hash.
+    # @return [Boolean] true if status is success.
+    def tx_succeeded?(hash)
+      tx_receipt = eth_get_transaction_receipt(hash)
+      !tx_receipt.nil? && !tx_receipt["result"].nil? && tx_receipt["result"]["status"] == "0x1"
```

Comment:
I think, It could be written as follows.
```suggestion
      !tx_receipt.nil? && tx_receipt.dig(:result, :status) == "0x1"
```

Reply:
you are right! nice catch. I will do the same with the `tx_mined?` method